projects/plugins/jetpack/modules/videopress/js/gutenberg-video-upload.js

```diff
@@ -1,9 +1,47 @@
 /* globals wp, lodash */
+window.videoPressUploadPoster = function ( guid, data ) {
+	// eslint-disable-next-line no-undef
+	return new Promise( function ( resolve, reject ) {
+		wp.media.ajax( 'videopress-get-upload-token', { async: true } ).done( function ( response ) {
+			// Set auth header with upload token.
+			var headers = {},
+				options = {};
+			var body = new FormData();
+			headers[ 'Authorization' ] =
+				'X_UPLOAD_TOKEN token="' +
+				response.upload_token +
+				'" blog_id="' +
+				response.upload_blog_id +
+				'"';
+			options.headers = headers;
+			options.method = 'POST';
+			options.url = 'https://public-api.wordpress.com/rest/v1.1/videos/' + guid + '/poster';
+
+			// Handle CORS.
+			options.credentials = 'omit';
+
+			Object.keys( data ).forEach( key => {
+				body.append( key, data[ key ] );
+			} );
+
+			options.body = body;
+
+			wp.apiFetch( options )
+				.then( function ( res ) {
+					resolve( res );
+				} )
+				.catch( function ( error ) {
+					reject( error );
+				} );
+		} );
+	} );
+};
+
 window.videoPressUploadTrack = function ( guid, kind, srcLang, label, vttFile ) {
 	// eslint-disable-next-line no-undef
 	return new Promise( function ( resolve, reject ) {
 		wp.media
-			.ajax( 'videopress-get-upload-token', { async: true, data: { filename: vttFile.name } } )
+			.ajax( 'videopress-get-upload-token', { async: true, data: { filename: vttFile.name } } ) // todo: maybe remove filename from here (not needed)
```

Comment:
Should it be removed ? (or should the comment be removed ? 😄 )

Reply:
I remember writing this `todo:` when mobbing. I think it should be removed but it should also be tested first before committing to make sure its not actually needed.